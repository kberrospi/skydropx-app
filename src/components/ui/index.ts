export * from './Formulario';
export * from './TablePrice';
