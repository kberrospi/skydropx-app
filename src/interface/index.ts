export * from './shipments'
export * from './dataForm'
export * from './dataPost'
export * from './labels'