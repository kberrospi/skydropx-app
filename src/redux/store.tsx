import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit'
import { labelSlice, shipmentSlice } from './reducers'

export function makeStore() {
  return configureStore({
    reducer: { 
      shipment: shipmentSlice.reducer,
      label: labelSlice.reducer
    },
  })
}

const store = makeStore()

export type AppState = ReturnType<typeof store.getState>

export type AppDispatch = typeof store.dispatch

export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  AppState,
  unknown,
  Action<string>
>

export default store