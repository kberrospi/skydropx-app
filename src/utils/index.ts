export * from './data-post'
export * from './helperForm'
export * from './helperLabels'