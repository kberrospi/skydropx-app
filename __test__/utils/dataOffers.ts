import { Included, Type } from "../../src/interface"

export const offer: Included[] = [
  {
      id: "376092",
      type: Type.Rates,
      attributes: {
          created_at: "2022-04-12T09:21:42.789-05:00",
          updated_at: "2022-04-12T09:21:48.914-05:00",
          amount_local: "155.0",
          currency_local: "MXN",
          provider: "FEDEX",
          service_level_name: "Fedex Express Saver",
          service_level_code: "FEDEX_EXPRESS_SAVER",
          service_level_terms: null,
          days: 5,
          duration_terms: null,
          zone: null,
          arrives_by: null,
          out_of_area: false,
          out_of_area_pricing: "0.0",
          total_pricing: "155.0",
          is_ocurre: false
      }
  },
  {
      id: "376093",
      type: Type.Rates,
      attributes: {
          created_at: "2022-04-12T09:21:42.793-05:00",
          updated_at: "2022-04-12T09:21:48.921-05:00",
          amount_local: "191.0",
          currency_local: "MXN",
          provider: "FEDEX",
          service_level_name: "Standard Overnight",
          service_level_code: "STANDARD_OVERNIGHT",
          service_level_terms: null,
          days: 2,
          duration_terms: null,
          zone: null,
          arrives_by: null,
          out_of_area: false,
          out_of_area_pricing: "0.0",
          total_pricing: "191.0",
          is_ocurre: false
      }
  },
  {
      id: "376096",
      type: Type.Rates,
      attributes: {
          created_at: "2022-04-12T09:21:48.635-05:00",
          updated_at: "2022-04-12T09:21:48.635-05:00",
          amount_local: "93.0",
          currency_local: "MXN",
          provider: "REDPACK",
          service_level_name: "Ecoexpress",
          service_level_code: "ECOEXPRESS",
          service_level_terms: null,
          days: 5,
          duration_terms: null,
          zone: null,
          arrives_by: null,
          out_of_area: true,
          out_of_area_pricing: "141.52",
          total_pricing: "234.52",
          is_ocurre: false
      }
  },
  {
      id: "376097",
      type: Type.Rates,
      attributes: {
          created_at: "2022-04-12T09:21:48.640-05:00",
          updated_at: "2022-04-12T09:21:48.640-05:00",
          amount_local: "145.0",
          currency_local: "MXN",
          provider: "REDPACK",
          service_level_name: "Express",
          service_level_code: "EXPRESS",
          service_level_terms: null,
          days: 2,
          duration_terms: null,
          zone: null,
          arrives_by: null,
          out_of_area: true,
          out_of_area_pricing: "174.36",
          total_pricing: "319.36",
          is_ocurre: false
      }
  },
  {
      id: "376100",
      type: Type.Rates,
      attributes: {
          created_at: "2022-04-12T09:21:48.877-05:00",
          updated_at: "2022-04-12T09:21:48.971-05:00",
          amount_local: "130.0",
          currency_local: "MXN",
          provider: "CARSSA",
          service_level_name: "Nacional",
          service_level_code: "NACIONAL",
          service_level_terms: null,
          days: 10,
          duration_terms: null,
          zone: null,
          arrives_by: null,
          out_of_area: false,
          out_of_area_pricing: "0.0",
          total_pricing: "130.0",
          is_ocurre: false
      }
  }
]