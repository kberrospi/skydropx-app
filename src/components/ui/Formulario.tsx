import { FC, useEffect } from 'react';
import { useRouter } from 'next/router';
import { Button, Input, Loading, Text, Grid } from '@nextui-org/react';
import { useForm } from '../../hooks/useForm';
import { formInit, handleSubmit, toServices } from '../../utils';
import { useAppDispatch, useAppSelector } from '../../redux/hook';
import styles from '../../styles/Form.module.css';


export const Formulario: FC = () => {

  const [ values, handleInputChange, reset ]  = useForm(formInit);
  const { from, to, width, height, weight, length } = values;
  const { status: statusShipment, data: dataShipment } = useAppSelector((state) => state.shipment);
  const dispatch = useAppDispatch();
  const router = useRouter();
  

  useEffect(() => {

    if( statusShipment === 'succeeded' ){

      toServices(dataShipment, router)
    }
    
  }, [dataShipment])
  

  return (
    
    <form 
      className={ styles.form } 
      onSubmit={ (e) => handleSubmit(e, values, dispatch) }
    >
      <Text h4 className={styles['title-sections']}> Datos de envío:  </Text>
      <Grid className={ styles['section-shipping'] } >
        <Input 
          className={ styles.input } 
          placeholder='Codigo postal Origen'
          label='Origen:'
          type='text'
          name='from'
          value={from}
          onChange={ handleInputChange }
        />
        <Input 
          className={ styles.input } 
          placeholder='Código postal Destino'
          label='Destino:'
          type='text'
          name='to'
          value={ to }
          onChange={ handleInputChange }
        />
      </Grid>

      <Text h4 className={styles['title-sections']}> Dimensiones del paquete: </Text>
      <Grid className={styles['section-package']} >
        <Input 
          className={ styles['input-container']}
          label='Largo'
          placeholder="Largo(CM)" 
          type='number' 
          name='length'
          value={ length || '' }
          onChange={ handleInputChange }
        />
        <Input  
          label='Ancho'
          placeholder="Ancho(CM)"  
          type='number'
          name='width'
          value={ width || '' }
          onChange={ handleInputChange }
        />
        <Input 
          label='Alto'
          placeholder="Alto(CM)" 
          type='number'
          name='height'
          value={ height || '' }
          onChange={ handleInputChange }
        />
        <Input 
          label='Peso'
          placeholder="Peso(KG)" 
          type='number' 
          name='weight'
          value={ weight || '' }
          onChange={ handleInputChange }
        />
      </Grid>
      <Button 
        className={ statusShipment === 'loading' ? '' : styles.button } 
        shadow 
        type='submit'  
        auto
        disabled={ statusShipment === 'loading' }
        
      > 
        { statusShipment === 'loading' 
          ? <Loading  color="currentColor" size="sm" />
          : 'Continuar'
        }
      
      </Button>
    </form>
   
  )
}
