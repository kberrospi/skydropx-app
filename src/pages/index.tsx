import { NextPage } from 'next'
import { useEffect } from 'react'
import { Layout } from '../components/layouts'
import { Formulario } from '../components/ui'
import { useAppDispatch } from '../redux/hook'
import { labelSlice, shipmentSlice } from '../redux/reducers'


const HomePage: NextPage = () => {

  const dispatch = useAppDispatch()
  const { reset } = shipmentSlice.actions
  const { resetLabel } = labelSlice.actions

  useEffect(() => {
    dispatch( reset() )
    dispatch( resetLabel() )
  }, [])
  

  return (
    <Layout title='Bienvenido a Skydropx'>
      <Formulario/>
    </Layout>
  )
}

export default HomePage
