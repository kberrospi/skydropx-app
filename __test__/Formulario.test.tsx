import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react'
import { Formulario } from '../src/components/ui';
import store from '../src/redux/store';
import "@testing-library/jest-dom";
import { Provider } from 'react-redux';
import { handleSubmit } from '../src/utils';



const state = store.getState()
store.dispatch = jest.fn();

/* jest.mock( '../src/utils', ()=> {
  handleSubmit: jest.fn()
} ) */

describe('Test en formulario', () => { 

  const wrapper = render(
    <Provider store={store}>
     <Formulario />
    </Provider>
  )

  beforeEach(() => {
    jest.clearAllMocks()
  })

  test('Debe mostrarse correctamente el formulario', async() => { 

    expect(wrapper).toMatchSnapshot()

   })

 /*  test('Debe llamar al submit', async() => { 

    fireEvent.click(wrapper.getByRole('button'))
    
   expect(handleSubmit).toHaveBeenCalledTimes(1)
  }) */

})