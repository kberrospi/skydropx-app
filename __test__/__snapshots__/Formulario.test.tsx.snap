// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en formulario Debe mostrarse correctamente el formulario 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="form"
      >
        <h4
          class="nextui-c-PJLV nextui-c-PJLV-ijXuRFq-css title-sections"
        >
           Datos de envío:  
        </h4>
        <div
          class="nextui-c-kRHeuF nextui-c-kRHeuF-igNCIse-css nextui-grid-item section-shipping"
        >
          <div
            class="nextui-c-ioOuFC nextui-c-ioOuFC-jTfqsF-helperColor-default nextui-c-ioOuFC-dpkacN-borderWeight-normal nextui-c-ioOuFC-kllHxD-size-md nextui-c-ioOuFC-iiIdOsz-css nextui-input-main-container nextui-input-main-container--normal"
            data-state="normal"
          >
            <label
              class="nextui-c-hzQjrs nextui-input-block-label"
              for="react-aria2142383709-1"
              id="react-aria2142383709-2"
            >
              Origen:
            </label>
            <div
              class="nextui-c-ghxhhI nextui-c-ghxhhI-fjttfO-isTextarea-false nextui-c-ghxhhI-iNKybL-animated-true nextui-input-container nextui-input-container--normal nextui-input-container--input input"
            >
              <div
                class="nextui-c-eXOOPO nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-ccrOPm-underlined-false nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--shadow"
              >
                <input
                  aria-labelledby="react-aria2142383709-2"
                  aria-multiline="false"
                  aria-placeholder="Codigo postal Origen"
                  aria-readonly="false"
                  aria-required="false"
                  autocomplete="off"
                  class="nextui-c-jeuecp nextui-input"
                  data-state="normal"
                  id="react-aria2142383709-1"
                  name="from"
                  placeholder="Codigo postal Origen"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="nextui-c-gJvMas nextui-input-helper-text-container"
            />
          </div>
          <div
            class="nextui-c-ioOuFC nextui-c-ioOuFC-jTfqsF-helperColor-default nextui-c-ioOuFC-dpkacN-borderWeight-normal nextui-c-ioOuFC-kllHxD-size-md nextui-c-ioOuFC-iiIdOsz-css nextui-input-main-container nextui-input-main-container--normal"
            data-state="normal"
          >
            <label
              class="nextui-c-hzQjrs nextui-input-block-label"
              for="react-aria2142383709-3"
              id="react-aria2142383709-4"
            >
              Destino:
            </label>
            <div
              class="nextui-c-ghxhhI nextui-c-ghxhhI-fjttfO-isTextarea-false nextui-c-ghxhhI-iNKybL-animated-true nextui-input-container nextui-input-container--normal nextui-input-container--input input"
            >
              <div
                class="nextui-c-eXOOPO nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-ccrOPm-underlined-false nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--shadow"
              >
                <input
                  aria-labelledby="react-aria2142383709-4"
                  aria-multiline="false"
                  aria-placeholder="Código postal Destino"
                  aria-readonly="false"
                  aria-required="false"
                  autocomplete="off"
                  class="nextui-c-jeuecp nextui-input"
                  data-state="normal"
                  id="react-aria2142383709-3"
                  name="to"
                  placeholder="Código postal Destino"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="nextui-c-gJvMas nextui-input-helper-text-container"
            />
          </div>
        </div>
        <h4
          class="nextui-c-PJLV nextui-c-PJLV-ijXuRFq-css title-sections"
        >
           Dimensiones del paquete: 
        </h4>
        <div
          class="nextui-c-kRHeuF nextui-c-kRHeuF-igNCIse-css nextui-grid-item section-package"
        >
          <div
            class="nextui-c-ioOuFC nextui-c-ioOuFC-jTfqsF-helperColor-default nextui-c-ioOuFC-dpkacN-borderWeight-normal nextui-c-ioOuFC-kllHxD-size-md nextui-c-ioOuFC-iiIdOsz-css nextui-input-main-container nextui-input-main-container--normal"
            data-state="normal"
          >
            <label
              class="nextui-c-hzQjrs nextui-input-block-label"
              for="react-aria2142383709-5"
              id="react-aria2142383709-6"
            >
              Largo
            </label>
            <div
              class="nextui-c-ghxhhI nextui-c-ghxhhI-fjttfO-isTextarea-false nextui-c-ghxhhI-iNKybL-animated-true nextui-input-container nextui-input-container--normal nextui-input-container--input input-container"
            >
              <div
                class="nextui-c-eXOOPO nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-ccrOPm-underlined-false nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--shadow"
              >
                <input
                  aria-labelledby="react-aria2142383709-6"
                  aria-multiline="false"
                  aria-placeholder="Largo(CM)"
                  aria-readonly="false"
                  aria-required="false"
                  autocomplete="off"
                  class="nextui-c-jeuecp nextui-input"
                  data-state="normal"
                  id="react-aria2142383709-5"
                  name="length"
                  placeholder="Largo(CM)"
                  type="number"
                  value=""
                />
              </div>
            </div>
            <div
              class="nextui-c-gJvMas nextui-input-helper-text-container"
            />
          </div>
          <div
            class="nextui-c-ioOuFC nextui-c-ioOuFC-jTfqsF-helperColor-default nextui-c-ioOuFC-dpkacN-borderWeight-normal nextui-c-ioOuFC-kllHxD-size-md nextui-c-ioOuFC-iiIdOsz-css nextui-input-main-container nextui-input-main-container--normal"
            data-state="normal"
          >
            <label
              class="nextui-c-hzQjrs nextui-input-block-label"
              for="react-aria2142383709-7"
              id="react-aria2142383709-8"
            >
              Ancho
            </label>
            <div
              class="nextui-c-ghxhhI nextui-c-ghxhhI-fjttfO-isTextarea-false nextui-c-ghxhhI-iNKybL-animated-true nextui-input-container nextui-input-container--normal nextui-input-container--input"
            >
              <div
                class="nextui-c-eXOOPO nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-ccrOPm-underlined-false nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--shadow"
              >
                <input
                  aria-labelledby="react-aria2142383709-8"
                  aria-multiline="false"
                  aria-placeholder="Ancho(CM)"
                  aria-readonly="false"
                  aria-required="false"
                  autocomplete="off"
                  class="nextui-c-jeuecp nextui-input"
                  data-state="normal"
                  id="react-aria2142383709-7"
                  name="width"
                  placeholder="Ancho(CM)"
                  type="number"
                  value=""
                />
              </div>
            </div>
            <div
              class="nextui-c-gJvMas nextui-input-helper-text-container"
            />
          </div>
          <div
            class="nextui-c-ioOuFC nextui-c-ioOuFC-jTfqsF-helperColor-default nextui-c-ioOuFC-dpkacN-borderWeight-normal nextui-c-ioOuFC-kllHxD-size-md nextui-c-ioOuFC-iiIdOsz-css nextui-input-main-container nextui-input-main-container--normal"
            data-state="normal"
          >
            <label
              class="nextui-c-hzQjrs nextui-input-block-label"
              for="react-aria2142383709-9"
              id="react-aria2142383709-10"
            >
              Alto
            </label>
            <div
              class="nextui-c-ghxhhI nextui-c-ghxhhI-fjttfO-isTextarea-false nextui-c-ghxhhI-iNKybL-animated-true nextui-input-container nextui-input-container--normal nextui-input-container--input"
            >
              <div
                class="nextui-c-eXOOPO nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-ccrOPm-underlined-false nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--shadow"
              >
                <input
                  aria-labelledby="react-aria2142383709-10"
                  aria-multiline="false"
                  aria-placeholder="Alto(CM)"
                  aria-readonly="false"
                  aria-required="false"
                  autocomplete="off"
                  class="nextui-c-jeuecp nextui-input"
                  data-state="normal"
                  id="react-aria2142383709-9"
                  name="height"
                  placeholder="Alto(CM)"
                  type="number"
                  value=""
                />
              </div>
            </div>
            <div
              class="nextui-c-gJvMas nextui-input-helper-text-container"
            />
          </div>
          <div
            class="nextui-c-ioOuFC nextui-c-ioOuFC-jTfqsF-helperColor-default nextui-c-ioOuFC-dpkacN-borderWeight-normal nextui-c-ioOuFC-kllHxD-size-md nextui-c-ioOuFC-iiIdOsz-css nextui-input-main-container nextui-input-main-container--normal"
            data-state="normal"
          >
            <label
              class="nextui-c-hzQjrs nextui-input-block-label"
              for="react-aria2142383709-11"
              id="react-aria2142383709-12"
            >
              Peso
            </label>
            <div
              class="nextui-c-ghxhhI nextui-c-ghxhhI-fjttfO-isTextarea-false nextui-c-ghxhhI-iNKybL-animated-true nextui-input-container nextui-input-container--normal nextui-input-container--input"
            >
              <div
                class="nextui-c-eXOOPO nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-ccrOPm-underlined-false nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--shadow"
              >
                <input
                  aria-labelledby="react-aria2142383709-12"
                  aria-multiline="false"
                  aria-placeholder="Peso(KG)"
                  aria-readonly="false"
                  aria-required="false"
                  autocomplete="off"
                  class="nextui-c-jeuecp nextui-input"
                  data-state="normal"
                  id="react-aria2142383709-11"
                  name="weight"
                  placeholder="Peso(KG)"
                  type="number"
                  value=""
                />
              </div>
            </div>
            <div
              class="nextui-c-gJvMas nextui-input-helper-text-container"
            />
          </div>
        </div>
        <button
          class="nextui-c-kZXRhs nextui-c-PJLV nextui-c-kZXRhs-ikltse-color-default nextui-c-kZXRhs-bWGFbV-size-md nextui-c-kZXRhs-cwXrJp-borderWeight-normal nextui-c-kZXRhs-fzQovo-shadow-true nextui-c-kZXRhs-exaRme-auto-true nextui-c-kZXRhs-gglxka-cv nextui-c-kZXRhs-jsIPTU-cv nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-kZXRhs-iPJLV-css nextui-button nextui-button--ready button"
          data-state="ready"
          type="submit"
        >
          <span
            class="nextui-button-text"
          >
            Continuar
          </span>
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="form"
    >
      <h4
        class="nextui-c-PJLV nextui-c-PJLV-ijXuRFq-css title-sections"
      >
         Datos de envío:  
      </h4>
      <div
        class="nextui-c-kRHeuF nextui-c-kRHeuF-igNCIse-css nextui-grid-item section-shipping"
      >
        <div
          class="nextui-c-ioOuFC nextui-c-ioOuFC-jTfqsF-helperColor-default nextui-c-ioOuFC-dpkacN-borderWeight-normal nextui-c-ioOuFC-kllHxD-size-md nextui-c-ioOuFC-iiIdOsz-css nextui-input-main-container nextui-input-main-container--normal"
          data-state="normal"
        >
          <label
            class="nextui-c-hzQjrs nextui-input-block-label"
            for="react-aria2142383709-1"
            id="react-aria2142383709-2"
          >
            Origen:
          </label>
          <div
            class="nextui-c-ghxhhI nextui-c-ghxhhI-fjttfO-isTextarea-false nextui-c-ghxhhI-iNKybL-animated-true nextui-input-container nextui-input-container--normal nextui-input-container--input input"
          >
            <div
              class="nextui-c-eXOOPO nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-ccrOPm-underlined-false nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--shadow"
            >
              <input
                aria-labelledby="react-aria2142383709-2"
                aria-multiline="false"
                aria-placeholder="Codigo postal Origen"
                aria-readonly="false"
                aria-required="false"
                autocomplete="off"
                class="nextui-c-jeuecp nextui-input"
                data-state="normal"
                id="react-aria2142383709-1"
                name="from"
                placeholder="Codigo postal Origen"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="nextui-c-gJvMas nextui-input-helper-text-container"
          />
        </div>
        <div
          class="nextui-c-ioOuFC nextui-c-ioOuFC-jTfqsF-helperColor-default nextui-c-ioOuFC-dpkacN-borderWeight-normal nextui-c-ioOuFC-kllHxD-size-md nextui-c-ioOuFC-iiIdOsz-css nextui-input-main-container nextui-input-main-container--normal"
          data-state="normal"
        >
          <label
            class="nextui-c-hzQjrs nextui-input-block-label"
            for="react-aria2142383709-3"
            id="react-aria2142383709-4"
          >
            Destino:
          </label>
          <div
            class="nextui-c-ghxhhI nextui-c-ghxhhI-fjttfO-isTextarea-false nextui-c-ghxhhI-iNKybL-animated-true nextui-input-container nextui-input-container--normal nextui-input-container--input input"
          >
            <div
              class="nextui-c-eXOOPO nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-ccrOPm-underlined-false nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--shadow"
            >
              <input
                aria-labelledby="react-aria2142383709-4"
                aria-multiline="false"
                aria-placeholder="Código postal Destino"
                aria-readonly="false"
                aria-required="false"
                autocomplete="off"
                class="nextui-c-jeuecp nextui-input"
                data-state="normal"
                id="react-aria2142383709-3"
                name="to"
                placeholder="Código postal Destino"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="nextui-c-gJvMas nextui-input-helper-text-container"
          />
        </div>
      </div>
      <h4
        class="nextui-c-PJLV nextui-c-PJLV-ijXuRFq-css title-sections"
      >
         Dimensiones del paquete: 
      </h4>
      <div
        class="nextui-c-kRHeuF nextui-c-kRHeuF-igNCIse-css nextui-grid-item section-package"
      >
        <div
          class="nextui-c-ioOuFC nextui-c-ioOuFC-jTfqsF-helperColor-default nextui-c-ioOuFC-dpkacN-borderWeight-normal nextui-c-ioOuFC-kllHxD-size-md nextui-c-ioOuFC-iiIdOsz-css nextui-input-main-container nextui-input-main-container--normal"
          data-state="normal"
        >
          <label
            class="nextui-c-hzQjrs nextui-input-block-label"
            for="react-aria2142383709-5"
            id="react-aria2142383709-6"
          >
            Largo
          </label>
          <div
            class="nextui-c-ghxhhI nextui-c-ghxhhI-fjttfO-isTextarea-false nextui-c-ghxhhI-iNKybL-animated-true nextui-input-container nextui-input-container--normal nextui-input-container--input input-container"
          >
            <div
              class="nextui-c-eXOOPO nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-ccrOPm-underlined-false nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--shadow"
            >
              <input
                aria-labelledby="react-aria2142383709-6"
                aria-multiline="false"
                aria-placeholder="Largo(CM)"
                aria-readonly="false"
                aria-required="false"
                autocomplete="off"
                class="nextui-c-jeuecp nextui-input"
                data-state="normal"
                id="react-aria2142383709-5"
                name="length"
                placeholder="Largo(CM)"
                type="number"
                value=""
              />
            </div>
          </div>
          <div
            class="nextui-c-gJvMas nextui-input-helper-text-container"
          />
        </div>
        <div
          class="nextui-c-ioOuFC nextui-c-ioOuFC-jTfqsF-helperColor-default nextui-c-ioOuFC-dpkacN-borderWeight-normal nextui-c-ioOuFC-kllHxD-size-md nextui-c-ioOuFC-iiIdOsz-css nextui-input-main-container nextui-input-main-container--normal"
          data-state="normal"
        >
          <label
            class="nextui-c-hzQjrs nextui-input-block-label"
            for="react-aria2142383709-7"
            id="react-aria2142383709-8"
          >
            Ancho
          </label>
          <div
            class="nextui-c-ghxhhI nextui-c-ghxhhI-fjttfO-isTextarea-false nextui-c-ghxhhI-iNKybL-animated-true nextui-input-container nextui-input-container--normal nextui-input-container--input"
          >
            <div
              class="nextui-c-eXOOPO nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-ccrOPm-underlined-false nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--shadow"
            >
              <input
                aria-labelledby="react-aria2142383709-8"
                aria-multiline="false"
                aria-placeholder="Ancho(CM)"
                aria-readonly="false"
                aria-required="false"
                autocomplete="off"
                class="nextui-c-jeuecp nextui-input"
                data-state="normal"
                id="react-aria2142383709-7"
                name="width"
                placeholder="Ancho(CM)"
                type="number"
                value=""
              />
            </div>
          </div>
          <div
            class="nextui-c-gJvMas nextui-input-helper-text-container"
          />
        </div>
        <div
          class="nextui-c-ioOuFC nextui-c-ioOuFC-jTfqsF-helperColor-default nextui-c-ioOuFC-dpkacN-borderWeight-normal nextui-c-ioOuFC-kllHxD-size-md nextui-c-ioOuFC-iiIdOsz-css nextui-input-main-container nextui-input-main-container--normal"
          data-state="normal"
        >
          <label
            class="nextui-c-hzQjrs nextui-input-block-label"
            for="react-aria2142383709-9"
            id="react-aria2142383709-10"
          >
            Alto
          </label>
          <div
            class="nextui-c-ghxhhI nextui-c-ghxhhI-fjttfO-isTextarea-false nextui-c-ghxhhI-iNKybL-animated-true nextui-input-container nextui-input-container--normal nextui-input-container--input"
          >
            <div
              class="nextui-c-eXOOPO nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-ccrOPm-underlined-false nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--shadow"
            >
              <input
                aria-labelledby="react-aria2142383709-10"
                aria-multiline="false"
                aria-placeholder="Alto(CM)"
                aria-readonly="false"
                aria-required="false"
                autocomplete="off"
                class="nextui-c-jeuecp nextui-input"
                data-state="normal"
                id="react-aria2142383709-9"
                name="height"
                placeholder="Alto(CM)"
                type="number"
                value=""
              />
            </div>
          </div>
          <div
            class="nextui-c-gJvMas nextui-input-helper-text-container"
          />
        </div>
        <div
          class="nextui-c-ioOuFC nextui-c-ioOuFC-jTfqsF-helperColor-default nextui-c-ioOuFC-dpkacN-borderWeight-normal nextui-c-ioOuFC-kllHxD-size-md nextui-c-ioOuFC-iiIdOsz-css nextui-input-main-container nextui-input-main-container--normal"
          data-state="normal"
        >
          <label
            class="nextui-c-hzQjrs nextui-input-block-label"
            for="react-aria2142383709-11"
            id="react-aria2142383709-12"
          >
            Peso
          </label>
          <div
            class="nextui-c-ghxhhI nextui-c-ghxhhI-fjttfO-isTextarea-false nextui-c-ghxhhI-iNKybL-animated-true nextui-input-container nextui-input-container--normal nextui-input-container--input"
          >
            <div
              class="nextui-c-eXOOPO nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-ccrOPm-underlined-false nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--shadow"
            >
              <input
                aria-labelledby="react-aria2142383709-12"
                aria-multiline="false"
                aria-placeholder="Peso(KG)"
                aria-readonly="false"
                aria-required="false"
                autocomplete="off"
                class="nextui-c-jeuecp nextui-input"
                data-state="normal"
                id="react-aria2142383709-11"
                name="weight"
                placeholder="Peso(KG)"
                type="number"
                value=""
              />
            </div>
          </div>
          <div
            class="nextui-c-gJvMas nextui-input-helper-text-container"
          />
        </div>
      </div>
      <button
        class="nextui-c-kZXRhs nextui-c-PJLV nextui-c-kZXRhs-ikltse-color-default nextui-c-kZXRhs-bWGFbV-size-md nextui-c-kZXRhs-cwXrJp-borderWeight-normal nextui-c-kZXRhs-fzQovo-shadow-true nextui-c-kZXRhs-exaRme-auto-true nextui-c-kZXRhs-gglxka-cv nextui-c-kZXRhs-jsIPTU-cv nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-kZXRhs-iPJLV-css nextui-button nextui-button--ready button"
        data-state="ready"
        type="submit"
      >
        <span
          class="nextui-button-text"
        >
          Continuar
        </span>
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
